internal/ethereum/blocklistener.go

```diff
@@ -381,22 +381,21 @@ func (bl *blockListener) handleNewBlock(mbi *minimalBlockInfo, addAfter *list.El
 // a recent block advertisement. So we need to work backwards to the last point of consistency with the current
 // chain and re-query the chain state from there.
 func (bl *blockListener) rebuildCanonicalChain() *list.Element {
-
-	log.L(bl.ctx).Debugf("Rebuilding in-memory canonical chain")
-
 	// If none of our blocks were valid, start from the first block number we've notified about previously
 	lastValidBlock := bl.trimToLastValidBlock()
 	var nextBlockNumber int64
 	var expectedParentHash string
 	if lastValidBlock != nil {
 		nextBlockNumber = lastValidBlock.number + 1
+		log.L(bl.ctx).Warnf("rebuildCanonicalChain partial rebuild from block %d", nextBlockNumber)
 		expectedParentHash = lastValidBlock.hash
 	} else {
 		firstBlock := bl.canonicalChain.Front()
 		if firstBlock == nil || firstBlock.Value == nil {
 			return nil
 		}
 		nextBlockNumber = firstBlock.Value.(*minimalBlockInfo).number
+		log.L(bl.ctx).Warnf("rebuildCanonicalChain brand new in-memory canonical chain from block %d", nextBlockNumber)
```

Comment:
These should not be warn levels as far as I know. This is normal operation as far as I'm aware

Reply:
I reverted other warn level logs to be info and left this one as `warn`. I think this one should be at `warn` level because we are throwing away the in memory chain here.